app/models/word.rb

```diff
@@ -1,7 +1,41 @@
 class Word
   include Mongoid::Document
-  field :word_name, type: String
+  include Mongoid::Attributes::Dynamic
+
+
+  belongs_to :syntactical_category, class_name: "Category", inverse_of: :syntactical_words, :autosave => true
+  has_and_belongs_to_many :semantical_categories, class_name: "Category", inverse_of: :semantical_words, :autosave => true
+
+  field :name, type: String
   field :description, type: String
-  field :syntactical_category, type: String
-  field :semantical-categories, type: String
+
+
+  # OPTIMIZE: if the word allready exist, it should ask the user if he wants to edit the existing word.
+  validates :name, uniqueness: {message: "Dieses Wort gibt es bereits." }
+
+  validates :name, presence: true;
+  validates :description, presence: true;
+
+
+
+  # those two functions are used for navigating from on word to an other.
+  # can't use the function index directly on mongo objects so i create the simple array words with all IDs to get the current index, afther that i use this index with a operatoin (+/- 1) to return the complete mongo object
+  def self.get_previous_word(current_word)
+
```

Comment:
This can get very slow as we have more and more words. Consider using a linked list (store next and previous word as a link in each word).  Or you could just access a random word. Word.all does not guarantee any order. Hence what you are doing here is just obtaining a pseudo random word.
Another problem is when you call this function on the first or the last word and get an index of -1